Comments appear as replies to root comment if replying on another page

I noticed the same but I thought I made a mistake. However, another stacker also noticed this now: https://stacker.news/items/245783

So maybe this is a real bug. Creating this ticket to investigate

Yes, good point. :) I am currently working on integrating `eslint` into our CI pipeline. I think there are some other, similar bugs. For example, I think scrolling collapsed comments into view no longer works consistently:

```
/home/ekzyis/programming/stacker.news/components/reply.js
   36:40  error  'onSuccess' is defined but never used
                                                                               no-unused-vars
   43:6   error  React Hook useEffect has missing dependencies: 'parentId' and 'replyOpen'. Either include them or remove the dependency array. If 'setReply' needs the current value of 'replyOpen', you can also s
witch to useReducer instead of useState and read 'replyOpen' in the reducer    react-hooks/exhaustive-deps
   93:41  error  'amount' is defined but never used
                                                                               no-unused-vars
   97:6   error  React Hook useCallback has missing dependencies: 'parentId' and 'replyOpen'. Either include them or remove the dependency array. If 'setReply' needs the current value of 'replyOpen', you can also
 switch to useReducer instead of useState and read 'replyOpen' in the reducer  react-hooks/exhaustive-deps
  102:6   error  React Hook useEffect has a missing dependency: 'replyOpen'. Either include it or remove the dependency array
                                                                               react-hooks/exhaustive-deps
```

The error at `97:6` is causing the bug of this ticket

--- a/components/reply.js
+++ b/components/reply.js
@@ -94,7 +94,7 @@ export default function Reply ({ item, onSuccess, replyOpen, children, placehold
     await upsertComment({ variables: { parentId, hash, hmac, ...values } })
     resetForm({ text: '' })
     setReply(replyOpen || false)
-  }, [upsertComment, setReply])
+  }, [upsertComment, setReply, parentId])
 
   const replyInput = useRef(null)
   useEffect(() => {

